x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/visualize_esql.ts

```diff
@@ -6,17 +6,39 @@
  */
 import { esFieldTypeToKibanaFieldType } from '@kbn/field-types';
 import type { ESQLSearchReponse } from '@kbn/es-types';
+import { validateQuery } from '@kbn/esql-validation-autocomplete';
+import { getAstAndSyntaxErrors } from '@kbn/esql-ast';
 import { VisualizeESQLUserIntention } from '@kbn/observability-ai-assistant-plugin/common/functions/visualize_esql';
 import { visualizeESQLFunction } from '../../common/functions/visualize_esql';
 import { FunctionRegistrationParameters } from '.';
 
+const getSystemMessage = (
+  intention: VisualizeESQLUserIntention,
+  query: string,
+  queryErrors: string[]
+) => {
+  if (queryErrors.length) {
+    return 'The query has syntax errors: ```\n' + queryErrors + '\n```';
+  }
+  return intention === VisualizeESQLUserIntention.executeAndReturnResults ||
+    intention === VisualizeESQLUserIntention.generateQueryOnly
+    ? 'These results are not visualized'
+    : 'Only following query is visualized: ```esql\n' + query + '\n```';
+};
+
 export function registerVisualizeESQLFunction({
   functions,
   resources,
 }: FunctionRegistrationParameters) {
   functions.registerFunction(
     visualizeESQLFunction,
     async ({ arguments: { query, intention }, connectorId, messages }, signal) => {
+      const { errors } = await validateQuery(query, getAstAndSyntaxErrors, {
```

Comment:
I will rerun the validation here (although I do it already in the respond function) as the user might click the Visualize query button and call the function manually.

Reply:
👍  mind adding a comment here?

Reply:
Done done https://github.com/elastic/kibana/commit/05398aceb2965f205104dd2dc30783b8c258702a